<?xml version="1.0"?>
<launch>
  	<!-- ================================ LEADER-FOLLOWER SETUP ================================ -->
	
	<!-- ================================ LEADER ================================ -->
	<arg name="cf_leader_name" value="demo_crazyflie10" />
	<!-- initial reference -->
	<arg name="cf_1_x" value="0  " /> 
	<arg name="cf_1_y" value="4  " />
	<arg name="cf_1_z" value="0.5" />

	<!-- =============================== FOLLOWER =============================== -->
	<arg name="cf_follower_name" value="demo_crazyflie9" />
	<!-- initial reference -->
  	<arg name="cf_2_x" value="-1" />
	<arg name="cf_2_y" value=" 4" />
	<arg name="cf_2_z" value=" 0.5" />
	

	<!-- Initialize global parameters -->
	<param name="cf_leader_name" type="string" value="$(arg cf_leader_name)" />
	<param name="cf_follower_name" type="string" value="$(arg cf_follower_name)" />

	<param name="cf_leader_x" type="double" value="$(arg cf_1_x)" />
	<param name="cf_leader_y" type="double" value="$(arg cf_1_y)" />
	<param name="cf_leader_z" type="double" value="$(arg cf_1_z)" />
	
	<param name="cf_follower_x" type="double" value="$(arg cf_2_x)" />
	<param name="cf_follower_y" type="double" value="$(arg cf_2_y)" />
	<param name="cf_follower_z" type="double" value="$(arg cf_2_z)" />

  	<!-- ================================ /LEADER-FOLLOWER SETUP ================================ -->

	<!-- ========================== ROS PARAMETERS ========================= -->
	<group ns="planning" >
		<rosparam command="load" file="$(find drones_rope_planning)/config/prob_definitiion.yaml" />
	</group>
	
	<group ns="obstacles" >
		<rosparam command="load" file="$(find drones_rope_planning)/config/cylinders_obstacles.yaml" />
	</group>
	
	<param name="waypoints_freq" type="double" value="40" />
	
	<param name="is_simulation"  value="0"/> 

	<!-- ========================== /ROS PARAMETERS ========================= -->

	<group ns="cf_leader">
		<node name="cf_controller" pkg="drones_control" type="hummingbird3_class.py"    args="$(arg cf_leader_name) 
			$(arg cf_1_x) $(arg cf_1_y) $(arg cf_1_z)" />
	
		<node name="leader_interface" pkg="execution" type="leader_traj_executor_position_controller_realtime.py"   args="$(arg cf_leader_name)"  />

	</group>

	<group ns="cf_follower">
		<node name="cf_controller" 	pkg="drones_control" type="hummingbird3_class.py"   args="$(arg cf_follower_name)  
			$(arg cf_2_x) $(arg cf_2_y) $(arg cf_2_z) " />
  		
    	<node name="follower_interface" pkg="execution" type="follower_traj_executor_position_controller_realtime.py" args="$(arg cf_follower_name) $(arg cf_leader_name) "   />
	</group>

	<!-- ================================== PLANNING STUFFF ================================== -->
	<group ns="planning" >
		<include file="$(find drones_rope_planning)/launch/realtime_moving_obstacles_demo.launch">
			 	<arg name="load_params"  value="0" />
		</include>
	</group>

</launch>
